app/api/__init__.py

```diff
@@ -26,11 +27,12 @@
 api.route(EventList, 'event_list', '/events')
 api.route(EventDetail, 'event_detail', '/events/<int:id>', '/tickets/<int:ticket_id>/event',
           '/microlocations/<int:microlocation_id>/event', '/social_links/<int:social_link_id>/event',
-          '/sponsors/<int:sponsor_id>/event')
+          '/sponsors/<int:sponsor_id>/event', '/tracks/<int:id>/event')
```

Comment:
Please correct if I'm wrong here.
According to the docs if I see 
```api.route(PersonDetail, 'person_detail', '/persons/<int:id>', '/computers/<int:computer_id>/owner')```

What I am trying to do here is to get Owner( which is a person ) of a specific. Now to get this I will define a method. I will define a method ``` def before_get_object(self, view_kwargs):``` to get the related id of Person with that specific computer. 
Here are you saying it just as ```id``` which could be interpreted as event's id and there is no such addition of this in ``` def before_get_object``` function EventDetail.

I tried this with sponsors and at this endpoint (```/sponsors/1/event```) and I got the details of the event with ID 1 but the sponsor with id 1 was associated with event with id 2